### PR TITLE
Change how gitlab-group is parsed on options

### PR DESCRIPTION
**Pull request by linuxgemini**
As talked in #637, this Draft PR revolves around this comment from @JoelSpeed: https://github.com/oauth2-proxy/oauth2-proxy/issues/637#issuecomment-650116230

## Description

Currently, `gitlab-group` is parsed in a way where it is comma seperated on the command flag but space seperated 

## Motivation and Context

The current documentation does not reflect this issue.

## How Has This Been Tested?

Aside from `go test` failing for Logging for HTTP requests (which is out of scope here but logged below):

```
[2020/06/26 20:04:04] [http.go:58] HTTP: listening on 127.0.0.1:4180
[2020/06/26 20:04:04] [http.go:60] HTTP: closing 127.0.0.1:4180
--- FAIL: TestLoggingHandler_ServeHTTP (0.00s)
    logging_handler_test.go:62: Log message was
        127.0.0.1 - - [2020/06/26 20:04:04] test-server GET - "/foo/bar" HTTP/1.1 "" 200 4 0.001

        instead of matching
        127.0.0.1 - - [2020/06/26 20:04:04] test-server GET - "/foo/bar" HTTP/1.1 "" 200 4 0.000
eyJfoobar.eyJfoobar.12345asdfFAIL
exit status 1
```

I have tested the result in an already-running Gitlab oauth2 test environment and haven't gotten any errors.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: providers/gitlab.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ import (
 type GitLabProvider struct {
 	*ProviderData
 
-	Group        string
+	Group        []string
```
Comment (JoelSpeed) [inline providers/gitlab.go:21]: Maybe rename this groups now that it is a slice?

File: providers/gitlab.go [hunk 2]
```diff
@@ -173,7 +173,7 @@ func (p *GitLabProvider) verifyGroupMembership(userInfo *gitlabUserInfo) error {
 	}
 
 	// Find a valid group that they are a member of
-	validGroups := strings.Split(p.Group, " ")
+	validGroups := p.Group
```
Comment (JoelSpeed) [inline providers/gitlab.go:176]: I don't think we need this variable do we, can we just inline it on the next line?

File: CHANGELOG.md [hunk 3]
```diff
@@ -52,9 +52,14 @@
   - Fixes an inconsistency in the `--exclude-logging-paths` option by renaming it to `--exclude-logging-option`.
   - This flag may now be given multiple times as with other list options
   - This flag also accepts comma separated values
+- [#639](https://github.com/oauth2-proxy/oauth2-proxy/pull/639) Change how gitlab-group is parsed on options
+  - Previously, the flag gitlab-group used comma seperated values, while the config option used space seperated values.
+  - This fixes the config value to use slices internally.
+  - The config option `gitlab_group` is now `gitlab_groups`
```
Comment (JoelSpeed) [inline CHANGELOG.md:58]:
```suggestion
  - The config option `gitlab_group` is now `gitlab_groups`
  - The environment variable `OAUTH2_PROXY_GITLAB_GROUP` is now `OAUTH2_PROXY_GITLAB_GROUPS`
```

**Comment by linuxgemini** [comment]
My pleasure

**Pull request merged**